lib/model/store.dart

```diff
@@ -993,22 +1009,34 @@ class UpdateMachine {
       }
     }
 
+    Future<void> updateZulipVersionData(ZulipVersionData data) async {
+      final account = globalStore.getAccount(accountId)!;
+      if (data.zulipVersion != account.zulipVersion
+          || data.zulipMergeBase != account.zulipMergeBase
+          || data.zulipFeatureLevel != account.zulipFeatureLevel) {
+        await globalStore.updateAccount(accountId, AccountsCompanion(
+          zulipVersion: Value(data.zulipVersion),
+          zulipMergeBase: Value(data.zulipMergeBase),
+          zulipFeatureLevel: Value(data.zulipFeatureLevel)));
+        connection.zulipFeatureLevel = data.zulipFeatureLevel;
+      }
+    }
+
     final stopwatch = Stopwatch()..start();
-    final initialSnapshot = await _registerQueueWithRetry(connection,
-      stopAndThrowIfNoAccount: stopAndThrowIfNoAccount);
+    InitialSnapshot? initialSnapshot;
+    try {
+      initialSnapshot = await _registerQueueWithRetry(connection,
+        stopAndThrowIfNoAccount: stopAndThrowIfNoAccount);
+    } on _ServerVersionUnsupportedException catch (e) {
+      connection.close();
+      await updateZulipVersionData(e.data);
```

Comment:
This will mutate `connection` after closing it. Probably doesn't throw, but incorrect in principle.

With the suggestion in the previous comment, this can inline `updateZulipVersionData`, and just skip the line that updates `connection`.

Maybe further clean up by having a method on `GlobalStore`:
```dart
if (!zulipVersionData.matchesAccount(globalStore.getAccount(accountId)!)) {
  await globalStore.updateZulipVersionData(accountId, zulipVersionData);
}
```